src/main/java/fr/skytasul/guardianbeam/Laser.java

```diff
@@ -391,15 +391,23 @@ protected void sendDestroyPackets(Player p) throws ReflectiveOperationException
 		@Override
 		public void moveStart(Location location) throws ReflectiveOperationException {
 			this.start = location;
-			if (main != null) moveFakeEntity(start, guardianID, guardian);
-		}
+			if (main != null) {
+        		initGuardian();
```

Comment:
There is no need to init the guardian each time, this can be optimized.

Reply:
Should only the createGuardianPacket be changed every time? Again, not exactly sure on how this resource works.